packages/module/src/WarningModal/WarningModal.tsx

```diff
@@ -13,11 +13,17 @@ export interface WarningModalProps extends Omit<ModalProps, 'ref'> {
   /** Whether modal requires a checkbox before confirming */
   withCheckbox?: boolean;
   /** Custom checkbox label */
-  checkboxLabel?: string;
+  checkboxLabel?: ReactNode;
   /** Visual variant of the confirm button */
   confirmButtonVariant?: ButtonVariant;
   /** Custom OUIA ID */
   ouiaId?: string | number;
+  /** Confirmation text input props */
+  confirmationInputProps?: TextInputProps;
+  /** Label for the text confirmation input */
+  confirmationInputLabel?: (deleteName?: string) => ReactNode;
```

Comment:
not sure if this one has to be a function - if the user of our component uses this prop, he can pass the confirmation text directly, not using another prop. I would just make it a ReactNode